### PR TITLE
backend/feat/split_settlement

### PR DESCRIPTION
**Pull request by 0utkarsh**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by srid** [comment]
Could you update this PR with latest commits from `main`? We now use github runners.

**Review by hkmangla — APPROVED** [review]

File: lib/mobility-core/src/Kernel/External/Payment/Interface/Types.hs [hunk 1]
```diff
@@ -159,6 +187,23 @@ data OrderStatusResp
   deriving stock (Show, Read, Eq, Generic)
   deriving anyclass (FromJSON, ToJSON, ToSchema)
 
+data SplitSettlementResponse = SplitSettlementResponse
+  { splitDetails :: [SplitDetailsResponse],
```
Comment (hkmangla) [inline lib/mobility-core/src/Kernel/External/Payment/Interface/Types.hs:191]: It should also Maybbe.

File: lib/mobility-core/src/Kernel/External/Payment/Juspay/Types/Common.hs [hunk 2]
```diff
@@ -158,7 +158,25 @@ data OrderData = OrderData
...
     amount_refunded :: Maybe Double,
-    refunds :: Maybe [RefundsData]
+    refunds :: Maybe [RefundsData],
+    split_settlement_response :: Maybe SplitSettlementResponse
+  }
+  deriving stock (Show, Generic)
+  deriving anyclass (FromJSON, ToJSON, ToSchema)
+
+data SplitSettlementResponse = SplitSettlementResponse
+  { split_details :: [SplitDetailsResponse],
```
Comment (hkmangla) [inline lib/mobility-core/src/Kernel/External/Payment/Juspay/Types/Common.hs:168]: Same here.

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: lib/mobility-core/src/Kernel/External/Payment/Interface/Types.hs [hunk 3]
```diff
@@ -71,9 +71,36 @@ data CreateOrderReq = CreateOrderReq
     mandateEndDate :: Maybe Text,
     metadataGatewayReferenceId :: Maybe Text,
     optionsGetUpiDeepLinks :: Maybe Bool,
-    metadataExpiryInMins :: Maybe Int
+    metadataExpiryInMins :: Maybe Int,
+    splitSettlementDetails :: Maybe SplitSettlementDetails
   }
 
+data Split = Split
```
Comment (khuzema786) [inline lib/mobility-core/src/Kernel/External/Payment/Interface/Types.hs:78]: @0utkarsh All this fields will always be sent by Juspay in SplitSettlementDetails ? Should we make it Maybe if not sure ?

**Review by khuzema786 — APPROVED** [review]

**Review by khuzema786 — APPROVED** [review]

**Review by khuzema786 — APPROVED** [review]

**Pull request merged**